Lib/idlelib/calltip.py

```diff
@@ -55,18 +55,46 @@ def refresh_calltip_event(self, event):
             self.open_calltip(False)
 
     def open_calltip(self, evalfuncs):
-        self.remove_calltip_window()
-
         hp = HyperParser(self.editwin, "insert")
         sur_paren = hp.get_surrounding_brackets('(')
         if not sur_paren:
+            # Not inside parentheses: Don't open a calltip, and close one if
+            # currently open.
```

Comment:
```suggestion
        if not sur_paren:  # Outer call just closed by releasing ).
```
See response to your response below.

Reply:
Typing ')' is not the only case where this branch will be reached, it can also happen in various other cases via `force_open_calltip_event()`.